pkg/kubernetes/status/readiness.go

```diff
@@ -0,0 +1,95 @@
+package status
+
+import (
+	"context"
+	"fmt"
+
+	kudoapi "github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+	label "github.com/kudobuilder/kudo/pkg/util/kudo"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// ComputeReadiness updates instance Status.Conditions.Ready based on current state of underlying resources
+// currently readiness examines the following types: Pods, StatefulSets, Deployments, ReplicaSets and DaemonSets
+// Instance is considered ready if all the resources linked to this instance are also ready
+func IsReady(i kudoapi.Instance, c client.Client) (ready bool, msg string, err error) {
+	resources, err := healthResources(c, i.Name, i.Namespace)
+	if err != nil {
+		return false, "", err
+	}
+	ready = true
+	readinessMessage := ""
+	for _, res := range resources {
+		healthy, _, err := IsHealthy(res)
+		if !healthy {
+			readinessMessage += fmt.Sprintf(", %v", err)
+			ready = false
+		}
+	}
+
+	return ready, readinessMessage, nil
+}
+
+func healthResources(c client.Client, instanceName, instanceNamespace string) ([]runtime.Object, error) {
+	instanceLabels, err := labels.Parse(fmt.Sprintf("%s=%s,%s=%s", label.InstanceLabel, instanceName, label.HeritageLabel, "kudo"))
+	if err != nil {
+		return nil, fmt.Errorf("unable to create list of labels to define health: %v", err)
+	}
+
+	dList := &appsv1.DeploymentList{}
```

Comment:
This seems somewhat repetitive. Wdyt about smth. like:

```suggestion
	lists := map[string]runtime.Object{
		"Deployment": &appsv1.DeploymentList{},
		"StatefulSet": &appsv1.StatefulSetList{},
		...
	}
	
	for typ, ll := range lists {
		err = c.List(context.TODO(), ll, &client.ListOptions{Namespace: instanceNamespace, LabelSelector: instanceLabels})
		if err != nil {
			return nil, fmt.Errorf("unable to pull resources of type %s: %v", err, typ)
		}
	}
```

Reply:
hmm yeah, but how do I do this then? 🤔 
```
for i := range ll.Items {
			result = append(result, &dList.Items[i])
		}
```

The Items is not on runtime.object. I would have to convert to unstructured and maybe somehow from that? 🤔  yeah I did not find an elegant solution so I am all ears :)

Reply:
ah, that's what it about. Yah, it can be done by using the typeMeta and casting but that's ugly. I guess it is what it is 🤷